Makefile

```diff
@@ -50,35 +96,35 @@ else
 endif
 
 setup-tests:
-	source $(conda_activate) base; conda create -y -n $(conda_env) --use-local \
+	source $(conda_activate) base; conda create -y -n $(conda_test_env) --use-local \
 		$(foreach x,$(extra_channels),-c $(x)) $(pkg_name) $(foreach x,$(test_pkgs),"$(x)") \
 		$(foreach x,$(extra_pkgs),"$(x)")
 
 conda-rerender: setup-build 
 	python $(workdir)/$(build_script) -w $(workdir) -l $(last_stable) -B 0 -p $(pkg_name) -r $(repo_name) \
-		-b $(branch) --do_rerender --conda_env $(conda_env) --ignore_conda_missmatch \
-		--conda_activate $(conda_activate)
+		-b $(branch) --do_rerender --conda_env $(conda_build_env) --ignore_conda_missmatch \
+		--conda_activate $(conda_activate) $(conda_build_extra)
 
 conda-build:
 	mkdir -p $(artifact_dir)
 
 	python $(workdir)/$(build_script) -w $(workdir) -p $(pkg_name) --build_version $(build_version) \
-		--do_build --conda_env $(conda_env) --extra_channels $(extra_channels) \
+		--do_build --conda_env $(conda_build_env) --extra_channels $(extra_channels) \
 		--conda_activate $(conda_activate) $(conda_build_extra)
 
 conda-upload:
-	source $(conda_activate) $(conda_env); \
+	source $(conda_activate) $(conda_build_env); \
 		anaconda -t $(conda_upload_token) upload -u $(user) -l $(label) --force $(artifact_dir)/*.tar.bz2
 
 conda-dump-env:
 	mkdir -p $(artifact_dir)
 
-	source $(conda_activate) $(conda_env); conda list --explicit > $(artifact_dir)/$(conda_env_filename).txt
+	source $(conda_activate) $(conda_test_env); conda list --explicit > $(artifact_dir)/$(conda_env_filename).txt
 
 run-tests:
-	source $(conda_activate) $(conda_env); python run_tests.py -H -v2 -n 1 `pwd`/tests/test_big_array.py
-	mv `pwd`/tests/test_big_array.py $(workdir)/
-	source $(conda_activate) $(conda_env); python run_tests.py -H -v2 --subdir
+	source $(conda_activate) $(conda_test_env); python run_tests.py -H -v2 -n 1 # `pwd`/tests/test_big_array.py
+	# mv `pwd`/tests/test_big_array.py $(workdir)/
+	# source $(conda_activate) $(conda_test_env); python run_tests.py -H -v2 --subdir
```

Comment:
@muryanto1 This was causing me issues.

Do we need to  move the test_big_array.py? Why do we call run_tests.py a second time, i don't think i understand what --subdir does.

Reply:
Regarding --subdir, I think some tests may be generating some temp files, and if we run tests with --subdir, tests will write those files in a temporary directory rather than under the repo dir.
Sorry, I cannot recall why we ran tests/test_big_array.py by itself first.

Reply:
I just made all the tests run, seeing no failures in circleci. We should be good then.